Topics/Software_Tools/Git/Automated_Deployment.md

```diff
@@ -0,0 +1,66 @@
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;However, most importantly, both continuous delivery and continuous deployment can only occur if the new changes pass all the tests from continuous integration. This means that in order to successfully implement continuous delivery or deployment, you need to have an extensive set of test for your code that can verify that the changes are suitable for these environments. Otherwise, continuous delivery or deployment can lead to issues within staging and testing environments, or, worse, issues for customers.
+ 
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;To summarize, automated deployment is used for continous deliver or deployment. In order for this to work, continuous integration needs to be used in order to extensively test the code and ensure it is acceptable to deploy to whichever environment (depending on if delivery or deployment is used). Although it requires a strong CI, it leads to faster development, less risky releases, and more customer feedback. 
+ <br>
+ <br>
+To learn more about CI/CD, please see the following article from Atlassian, an established software company from Australia: https://www.atlassian.com/continuous-delivery/principles/continuous-integration-vs-delivery-vs-deployment
+ - To learn more about the difference between continous delivery and continuous deployment, see the following article: https://docs.github.com/en/actions/deployment/about-deployments/deploying-with-github-actions
+
+______
+
+ ### Examples of Common Automated Deployment Software
+ 
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Atlassian recommends using Bitbucket Pipelines (https://bitbucket.org/product/features/pipelines), which is their "CI/CD tool that's integrated into Bitbucket Cloud," their version of GitHub (https://bitbucket.org/product/). BitBucket provides extended CI/CD features that allow you to integrate with Jira, a task management software by Atlassian. However, you can implement CI/CD fairly easily with GitHub Actions.
+  <br>
+  <br>
+ ### How to Set Up Automated Deployment with GitHub Actions
+ 
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; In order to set up automated deployment, you need to know the following: when do you want deployments to occur (on pull request, on push, on workflow dispatch, etc), where do you want the deployment to be (staging or production, continuous delivery vs continuous deployment), and what are the requirements for your environment (what packages do you need to set install or software do you need to set up to deploy).
+
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Once you know these things, you can start setting up your GitHub Action using a yaml file. Start with the `on:` command and put all the situations where you want deployment to occur (and on what branches). Then, under `jobs:`, create a `deployment:` section and set the environment, what os it runs on, and the steps necessary to deploy.
```

Comment:
Linking to a simple workflow here to serve as a motivating example would be useful for readers. You could move [this article you linked below](https://techblog.geekyants.com/github-actions-for-automating-builds-for-your-app) up here since it features an example workflow.

also: capitalize "os" or replace with "operating system"

Reply:
Done, I moved the link you referred to to the top of the section